components/scream/src/mct_coupling/scream_cxx_f90_interface.cpp

```diff
@@ -82,12 +89,67 @@ void scream_init (const MPI_Fint& f_comm,
   enable_fpes(fpe_mask);
+
+  //                 SCORPIO                        //
+  // Create the set of SCORPIO output files and their respective
+  // dimensions and variables.
+  eam_init_pio_subsystem(f_comm,compid,false);   // Gather the initial PIO subsystem data creater by component coupler
+  // Register the set of output files:
+  register_outfile("example_pio_structured_v2.nc");
+  // Register the set of dimensions per output file
+  register_dimension("example_pio_structured_v2.nc","x","horizontal distance",10);
+  register_dimension("example_pio_structured_v2.nc","y","vertical distance",3);
+  register_dimension("example_pio_structured_v2.nc","z","height",2);
+  register_dimension("example_pio_structured_v2.nc","time","time",0);
+  // Register the set of variables per output file
+  std::string vec_time[] = {"time"};
+  std::string vec_x[]    = {"x"};
+  std::string vec_y[]    = {"y"};
+  std::string vec_z[]    = {"z"};
+  std::string vec_xt[]   = {"x","time"}; 
+  std::string vec_yt[]   = {"y","time"}; 
+  std::string vec_xyt[]  = {"x","y","time"}; 
+  std::string vec_yxt[]  = {"y","x","time"}; 
+  std::string vec_xyzt[] = {"x","y","z","time"}; 
+  register_variable("example_pio_structured_v2.nc","time","time",1,vec_time, PIO_REAL,"t");
+  register_variable("example_pio_structured_v2.nc","x","answer to space and time",1,vec_x, PIO_REAL,"x-real");
+  register_variable("example_pio_structured_v2.nc","y","answer to space and time",1,vec_y, PIO_REAL,"y-real");
+  register_variable("example_pio_structured_v2.nc","z","answer to space and time",1,vec_z, PIO_REAL,"z-real");
+  register_variable("example_pio_structured_v2.nc","bar","answer to space and time",3,vec_xyt, PIO_REAL,"xyt-real");
+  register_variable("example_pio_structured_v2.nc","foo","answer to space and time",3,vec_yxt, PIO_REAL,"yxt-real");
+  register_variable("example_pio_structured_v2.nc","bar_flip","answer to space and time",3,vec_xyt, PIO_REAL,"xyt-real");
+  register_variable("example_pio_structured_v2.nc","foo_flip","answer to space and time",3,vec_yxt, PIO_REAL,"yxt-real");
+  register_variable("example_pio_structured_v2.nc","foo_big","answer to space and time",4,vec_xyzt, PIO_REAL,"xyzt-real");
+
+  eam_pio_enddef("example_pio_structured_v2.nc");
+
+  std::array<Real,10> x_data;
+  std::array<Real, 3> y_data;
+  std::array<Real, 2> z_data = {100,200};
+  Int m_rank; 
+  MPI_Comm_rank(mpi_comm_c,&m_rank);
+  for (int ii=0;ii<10;ii++) {
+    x_data[ii] = ii + 1.0;
+  }
+  for (int jj=0;jj<3;jj++) {
+    y_data[jj] = jj + 1.0;
+  }
+  std::array<Int,1> xdim = {10};
+  std::array<Int,1> ydim = {3};
+  std::array<Int,1> zdim = {2};
+  grid_write_data_array("example_pio_structured_v2.nc","x",xdim,ekat::util::data(x_data));
+  grid_write_data_array("example_pio_structured_v2.nc","y",ydim,ekat::util::data(y_data));
+  grid_write_data_array("example_pio_structured_v2.nc","z",zdim,ekat::util::data(z_data));
+  
+  //                 SCORPIO DONE                    //
 }
```

Comment:
I'm surprised that actual work is occurring in an f90_interface file and within an extern C block - I thought that stuff was just for linking C++ we wrote elsewhere with F90. I'm not objecting, I'd just like to understand. Is this temporary, or will we always initialize and write all of our output from this interface layer (at least until we get rid of that F90 layer)?

Reply:
oh no, good catch.  This was supposed to be deleted.  I guess I missed it.  Will fix.